### ISSUE TITLE
Casting to chromecast stopped working in 10.6.2

### ISSUE THREAD

**Issue by TheTerrasque** [jellyfin/jellyfin#3852]
**Describe the bug**
 Casting video to Chromecast does not work any more on my setup on version 10.6.2

**System (please complete the following information):**
 - OS: Windows 10
 - Virtualization: None
 - Clients: Browser, Android
 - Browser: Chrome 84.0.4147.105
 - Jellyfin Version: 10.6.2
 - Playback: Seem to be same on all
 - Installed Plugins: OpenSubtitles
 - Reverse Proxy: traefik
 - Base URL: /
 - Networking: explained later
 - Storage: local

**To Reproduce**
* Connect to Chromecast from either browser or android app
* After connected and Jellyfin running on chromecast:
* Click play on media file
* See UI reacting to click, but no response happen, as if it wasn't clicked

**Expected behavior**
* Video plays on chromecast, or at least give an error

**Logs**
* No relevant entry in logs. 
* Log level : info
* Chrome browser once showed an error indicating timeout communicating with Chromecast

**Additional context**
Network setup:
* Jellyfin runs on Windows desktop, Chromecast is in same /24 net
* Traefik 1.7 is running on Raspberry pi, handling incoming on port 80 and 443
* Reverse proxy from domain to Jellyfin's internal ip and port is set up on traefik
* SSL cert is generated by Let's Encrypt, 
* Android app configured against external domain over https
Tried:
* Restarted Jellyfin server
* Tested with Chrome browser using localhost and external domain over https, no video in either cases.
* In-browser playback worked fine
* Tested from Android app, same result.
* In-app playback worked fine on Android
* Tested on two different Chromecast devices, same result.

Problem started after upgrading to 10.6.2, downgrading Jellyfin to 10.6.1 fixed it again. 

There is a fix for "inverted logic for LAN IP detection" in 10.6.2. I'm guessing fixing that have triggered another bug, possibly related to the network configuration I'm running.

**Comment by AtomicPumpkin** [comment]
Also experiencing the same issue with 10.6.2, my setup is essentially the same except I have Nginx handing reverse proxy duties.

**Comment by YouKnowBlom** [comment]
is this on the unstable chromecast branch?

**Comment by sybnex** [comment]
same here...

**Comment by idittansikte** [comment]
Same here. I need to use lan IP again instead of domain name via reverse proxy. Was not working some time before 10.6.1, started working 10.6.1 and now it's not working again in 10.6.2.

**Comment by Harteex** [comment]
Casting wouldn't work for me because Jellyfin sends Chromecast my local IP and port. In my reverse proxy setup the connection must go through the reverse proxy.

I fixed it for me by modifying jellyfin-web\plugins\chromecastPlayer\plugin.js
Changed "message.serverAddress = serverAddress" to "message.serverAddress = message.serverAddress" (so that it ignores the serverAddress from chromecastHelper)

Might be a good idea to at least let the user configure to always use the external address that the caster uses.

**Comment by dpraul** [comment]
I rolled back to 10.6.1 and this was still an issue for me, so not certain that I'm suffering the same issue as everyone else, but I found a (very ugly) workaround for my issue!

I stumbled on [this reddit thread](https://www.reddit.com/r/jellyfin/comments/gkui48/unable_to_cast_from_android_app_to_chromecast/) where someone was encountering a similar issue, and they figured out that the Chromecast plugin uses `LocalAddress` from `/System/Info/Public`. I'm fairly new to Jellyfin but I've seen people mention you used to be able to configure what this would respond with, but those options seem to be gone now.

Anyway, they suggested a terrible nginx hack of just rewriting the response. I threw this into my nginx config and suddenly Chromecast is working again!

```
location /System/Info/Public {
    add_header Content-Type "application/json; charset=utf8";
    return 200 '{"LocalAddress": "https://jellyfin.mydomain.com","ServerName": "MyServerName","Version": "10.6.2","ProductName": "Jellyfin Server","OperatingSystem": "Linux","Id": "f493033cefec41388dd38783a66f50cc"}';
}

**Comment by h0od** [comment]
I also starting to have this problem a week ago. It seem to have been introduced shortly after upgrading to 10.6.2, but I can't tell for sure if it happened immediately after upgrading. And like @dpraul, I rolled back to 10.6.1 and the problem is still there, which confuses me. I'm running Jellyfin in a Docker image (linuxserver), with a nginx as reverse proxy. There's HTTPS (Letsencrypt) on the proxy, and plain HTTP between proxy and Jellyfin. It has worked flawlessly for months, and my Chromecast have not had any problems with it. My external URL is like this: `https://jellyfin.mydomain.com`

Like @dpraul, I also found this [Reddit thread](https://www.reddit.com/r/jellyfin/comments/gkui48/unable_to_cast_from_android_app_to_chromecast/). Reading that, I got suspicious about my `LocalAddress` field in the response referring to Jellyfin's internal Docker IP, which of course is not accessible externally, not even on my own LAN where I have my Chromecast.

To solve this temporarily, I did almost exactly like @dpraul suggested and made a ugly hack to my nginx config to indicate the external URL in the `LocalAddress` field:
```
    location /System/Info/Public {
        default_type "application/json; charset=utf-8";
        alias /etc/nginx/private/jellyfin_info_public.json;
    }
```
The `/etc/nginx/private/jellyfin_info_public.json` file contains the json response with the external URL. After this little hack, casting to Chromecast works again.

So apparently, the Chromecast Receiver app is not getting the correct address. So the error is probably in the Cast Sender end, which is the Android app in my case.

**Comment by benja-opazo** [comment]
I started having the same problem, and inspired by this thread I applied this quick fix with apache

```
        <Location "/System/Info/Public">
                AddOutputFilterByType SUBSTITUTE application/json
                Substitute "s|<your_local_addres>|<your_public_address>|ni"
        </Location>
```

It is now working

**Comment by mholt** [comment]
I believe this was introduced in 10.6 (as I am also experiencing it with 10.6.0).

Unlike others in this thread, my `/System/Info/Public` does not have any `LocalAddress` at all:

```
{"ServerName":"MyServerName","Version":"10.6.0","ProductName":"Jellyfin Server","OperatingSystem":"Linux","Id":"f493033cefec41388dd38783a66f50cc"}
```

So I just added it (similar to other workarounds above). Here is a Caddy config (Caddyfile) excerpt that worked for me:

```
handle /media/System/Info/Public {
	header Content-Type "application/json; charset=utf8"
	respond `{"LocalAddress":"http://10.0.1.2","ServerName":"MyServerName","Version":"10.6.0","ProductName":"Jellyfin Server","OperatingSystem":"Linux","Id":"f493033cefec41388dd38783a66f50cc"}`
}
```

**Comment by dkanada** [comment]
Can anyone else confirm `LocalAddress` is the culprit here and see if any input for that setting fixes Chromecast devices?

> This reminds me of the Chromecast bug where Chromecast hard-codes its DNS servers as 8.8.8.8 and so we need a way to configure a different external address when accessing Jellyfin locally...

This is actually the reason we added that option, to fix Google DNS issues without requiring other tools or software.

**Comment by bialad** [comment]
I've done the same think in my setup, overriding LocalAddress in my reverse proxy solves the issue for me.

**Comment by chenichadowitz** [comment]
I've also done the same thing, statically overriding LocalAddress in my nginx reverse proxy.

**Comment by sybnex** [comment]
When I check the url at /System/Info/Public, I got the IP address from the internal docker network. 

`{"LocalAddress":"http://172.17.0.2:8096","ServerName":"jellyfin","Version":"10.6.1","ProductName":"Jellyfin Server","OperatingSystem":"Linux","Id":"c424b4cf1cc545a9bce6c3df21e49bfc"}`

But with jellyfin 10.6.1 it is working intern as well as extern.

When I used 10.6.2, casting was not working anymore.

**Comment by anthonylavado** [comment]
Possible link to #3712, with origins in #3033.

**Comment by bengalih** [comment]
> Possible link to #3712, with origins in #3033.

I opened #3712 and just saw this linked.  Would really like some answers here because none of this is seemingly making sense to the end users.  Several comments above mention hacking the information returned  `/media/System/Info/Public` to advertise the actual public IP/DNS instead of the local address that JF is providing.  They stated that this fixed their problem.

On the surface this would appear to make sense, but if you read my issues and eventual resolution on #3712, you will see that my configuration is currently working perfectly fine with the `/media/System/Info/Public` showing my local (10.x.x.x) IP.

When I say "working perfectly" I mean I am able to not only cast to my chromecast devices on my internal network, but I am also able to cast from a remote device (like my phone) to a CC device remote from my home network (i.e. local to wherever my phone is).  My JF setup is a straight Windows install which is proxied through an NGINX server (running on embedded linux) and *only* port 443 is accessible to the outside (i.e. port 80 is actually blocked by my ISP).

How is it possible that even though my `/System/Info/Public` returns a local 10.x.x.x address JF seemingly works without issue while others seemingly are forced to hack in their public address.

If the devs would provide some answers to these questions it would really help us users troubleshoot these issues and provide better feedback to you all.

**Comment by dpraul** [comment]
> How is it possible that even though my `/System/Info/Public` returns a local 10.x.x.x address JF seemingly works without issue while others seemingly are forced to hack in their public address.

AFAIK the jellyfin chromecast connector will leverage the `LocalAddress` returned from `/System/Info/Public` when it detects that your client is on the same network. So the hacks in here are _probably_ only necessary when casting within the local network.

**Comment by bengalih** [comment]
> > How is it possible that even though my `/System/Info/Public` returns a local 10.x.x.x address JF seemingly works without issue while others seemingly are forced to hack in their public address.
> 
> AFAIK the jellyfin chromecast connector will leverage the `LocalAddress` returned from `/System/Info/Public` when it detects that your client is on the same network. So the hacks in here are _probably_ only necessary when casting within the local network.

Ok, I'll buy that.  Curious exactly how this works then with remote.
Curious too - you don't mention from your fix above how your JF is working...are you using a docker container as well?
It seems most people still having this issue are in that configuration, if not I'd be curious to know why you needed the hack and I didn't.

**Comment by benja-opazo** [comment]
> So the hacks in here are probably only necessary when casting within the local network.

I had the opposite behavior, when casting within the local network chromecast worked as expected, but when casting outside my network was when problems appeared. Fixing the local address in ```/System/Info/Public``` fixed the casting outside my network, and casting inside my network continued working as expected

**Comment by dpraul** [comment]
> Curious too - you don't mention from your fix above how your JF is working...are you using a docker container as well?
> It seems most people still having this issue are in that configuration, if not I'd be curious to know why you needed the hack and I didn't.

Yes, mine is in docker - the difference is probably in docker networks. Mine was in a private subnetwork (I think the term is overlay?), so the IP was only accessible by other containers in the network (which my chromecast device is not). If you're running in bridge mode (which I believe is the default) and have the ports mounted, you can access the IP directly.

So what that means is, when the chromecast connector determines you _are_ on the local network, it will use the internal IP and internal port - otherwise, it uses the external address and external port. So chromecast should work as long as the container is using the bridge network and has the ports mapped.

**Comment by BaronGreenback** [comment]
Any chance one of you could test Docker image crobibero/jellyfin:3257? (Just asked for it to be rebuilt).

Details here:- https://github.com/jellyfin/jellyfin/pull/3994/

If it doesn't work "out of the box" could you enable network logging (details on the PR) and send me the info which says how it's selected the bind address?

It also implements functionality which would make chromecast work - but it would be nice to get it working with no setup.

**Comment by mholt** [comment]
Would that solution only work with Docker then? I don't use Docker but still have this problem.

**Comment by BaronGreenback** [comment]
No - if you can compile the code - please feel free to give it a go. It's the latest unstable as of yesterday, with a new network backend.

**Comment by Pegasystem** [comment]
For anyone Googling this and using Traefik 2, here's a .yml file to fix the issue:

```
http:
  routers:
    chromecast-rtr:
      rule: "HostHeader(`your.jellyfin.url`) && Path(`/System/Info/Public`)"  # Replace with the proper URL
      entryPoints:
        - https
      service: jellyfin-chromecast-local
      priority: 100  # Set this to anything that's higher than your normal Jellyfin rule - if it doesn't have a priority, add one
  services:
    jellyfin-chromecast-local:
      loadBalancer:
        servers:
          - url: "http://webserver-hosting-fixed-json-file/"  # Replace with the proper URL
```

Unfortunately Traefik isn't able to serve files on it's own from what I could find, so you'll have to host the modified json file somewhere.

**Comment by BaronGreenback** [comment]
As said - 3394 can definitely sort it by adding something like below to the settings - it would just be nice to have it working by default.

<PublishedServerUriBySubnet>
<string>192.168.1.0/24=net1access:1270</string>
<string>192.168.2.1/24=net2access</string>
<string>external=jellyfinoutside.com</string>
</PublishedServerUriBySubnet>

**Comment by sybnex** [comment]
> Any chance one of you could test Docker image crobibero/jellyfin:3257? (Just asked for it to be rebuilt).
> 
> Details here:- #3994
> 
> If it doesn't work "out of the box" could you enable network logging (details on the PR) and send me the info which says how it's selected the bind address?
> 
> It also implements functionality which would make chromecast work - but it would be nice to get it working with no setup.

does not start:

`[18:54:30] [ERR] [1] Jellyfin.Server.Migrations.MigrationRunner: Could not apply migration 'MigrateDisplayPreferencesDatabase'
Microsoft.EntityFrameworkCore.DbUpdateException: An error occurred while updating the entries. See the inner exception for details.
 ---> Microsoft.Data.Sqlite.SqliteException (0x80004005): SQLite Error 19: 'FOREIGN KEY constraint failed'.
   --- End of inner exception stack trace ---
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.Execute(IRelationalConnection connection)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.Execute(IEnumerable`1 commandBatches, IRelationalConnection connection)
   at Microsoft.EntityFrameworkCore.Storage.RelationalDatabase.SaveChanges(IList`1 entries)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChanges(IList`1 entriesToSave)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChanges(DbContext _, Boolean acceptAllChangesOnSuccess)
   at Microsoft.EntityFrameworkCore.Storage.Internal.NoopExecutionStrategy.Execute[TState,TResult](TState state, Func`3 operation, Func`3 verifySucceeded)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChanges(Boolean acceptAllChangesOnSuccess)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChanges(Boolean acceptAllChangesOnSuccess)
   at Jellyfin.Server.Migrations.Routines.MigrateDisplayPreferencesDb.Perform()
   at Jellyfin.Server.Migrations.MigrationRunner.Run(CoreAppHost host, ILoggerFactory loggerFactory)
`

**Comment by BaronGreenback** [comment]
That would be https://github.com/jellyfin/jellyfin/issues/4009 (problem in the unstable @ startup)

Could you try running it again - and see it it works this time.

**Comment by sybnex** [comment]
> That would be #4009 (problem in the unstable @ startup)
> 
> Could you try running it again - and see it it works this time.

It is still the same "old" image: https://hub.docker.com/r/crobibero/jellyfin/tags 
Do you create a new tag or is the pipeline not finished yet?

**Comment by crobibero** [comment]
What's linked is an open image, not an issue with the networking change.

**Comment by BaronGreenback** [comment]
same image crobibero/jellyfin:3257 - PR changes as i screwed up git.

**Comment by sybnex** [comment]
`sudo docker run -d \
	--restart always \
	--name jellyfin \
	--volume "/volume1/docker/jellyfin/config:/config" \
	--volume "/volume1/docker/jellyfin/cache:/cache" \
	--volume "/volume1/video:/media:ro" \
	-p 8096:8096 \
	--device /dev/dri/renderD128 \
	--device /dev/dri/card0 \
	--memory 3g \
	crobibero/jellyfin:3257
`

[19:18:01] [ERR] [1] Jellyfin.Server.Migrations.MigrationRunner: Could not apply migration 'MigrateDisplayPreferencesDatabase'
Microsoft.EntityFrameworkCore.DbUpdateException: An error occurred while updating the entries. See the inner exception for details.
 ---> Microsoft.Data.Sqlite.SqliteException (0x80004005): SQLite Error 19: 'FOREIGN KEY constraint failed'.
   at Microsoft.Data.Sqlite.SqliteException.ThrowExceptionForRC(Int32 rc, sqlite3 db)
   at Microsoft.Data.Sqlite.SqliteDataReader.NextResult()
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChanges(Boolean acceptAllChangesOnSuccess)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChanges(Boolean acceptAllChangesOnSuccess)
   at Jellyfin.Server.Migrations.Routines.MigrateDisplayPreferencesDb.Perform()
   at Jellyfin.Server.Migrations.MigrationRunner.Run(CoreAppHost host, ILoggerFactory loggerFactory)
[19:18:01] [FTL] [1] Main: Error while starting server.
Microsoft.EntityFrameworkCore.DbUpdateException: An error occurred while updating the entries. See the inner exception for details.
 ---> Microsoft.Data.Sqlite.SqliteException (0x80004005): SQLite Error 19: 'FOREIGN KEY constraint failed'.
   at Microsoft.Data.Sqlite.SqliteException.ThrowExceptionForRC(Int32 rc, sqlite3 db)
   at Microsoft.Data.Sqlite.SqliteDataReader.NextResult()
   at Microsoft.Data.Sqlite.SqliteCommand.ExecuteReader(CommandBehavior behavior)
[19:18:01] [INF] [1] Emby.Server.Implementations.ApplicationHost: Disposing CoreAppHost
[19:18:01] [INF] [1] Emby.Server.Implementations.ApplicationHost: Disposing SessionWebSocketListener
[19:18:01] [INF] [1] Emby.Server.Implementations.ApplicationHost: Disposing EmbyTV
[19:18:01] [INF] [2] Main: Received a SIGTERM signal, shutting down

**Comment by BaronGreenback** [comment]
Can you rename your displaypreferences.db to *.old and try again?

**Comment by sybnex** [comment]
he don't like the empty database 

[19:30:26] [INF] [1] Jellyfin.Server.Migrations.MigrationRunner: Applying migration 'MigrateDisplayPreferencesDatabase'
[19:30:26] [ERR] [1] Jellyfin.Server.Migrations.MigrationRunner: Could not apply migration 'MigrateDisplayPreferencesDatabase'
Error: SQLitePCL.pretty.SQLiteException: no such table: userdisplaypreferences
   at SQLitePCL.pretty.SQLiteException.Throw(Int32 rc, Int32 extended, String msg)
   at SQLitePCL.pretty.SQLiteException.CheckOk(sqlite3 db, Int32 rc)
   at SQLitePCL.pretty.SQLiteDatabaseConnection.PrepareStatement(String sql, String& tail)
   at SQLitePCL.pretty.DatabaseConnection.PrepareStatement(IDatabaseConnection This, String sql)
   at SQLitePCL.pretty.DatabaseConnection.<>c__DisplayClass5_0.<Query>b__0()
   at SQLitePCL.pretty.DelegatingEnumerable`1.GetEnumerator()
   at Jellyfin.Server.Migrations.Routines.MigrateDisplayPreferencesDb.Perform()
   at Jellyfin.Server.Migrations.MigrationRunner.Run(CoreAppHost host, ILoggerFactory loggerFactory)
[19:30:26] [FTL] [1] Main: Error while starting server.

**Comment by BaronGreenback** [comment]
Open up the migrations.xml (same folder as the settings) and add

    <ValueTupleOfGuidString>
      <Item1>06387815-c3cc-421f-a888-fb5f9992bea8</Item1>
      <Item2>MigrateDisplayPreferencesDatabase</Item2>
    </ValueTupleOfGuidString>

**Comment by sybnex** [comment]
ok it is starting now ... but chromecast still produce an error:

`[2020-08-31 19:41:26.347 +00:00] [WRN] [20] Jellyfin.Server.Middleware.ResponseTimeMiddleware: Slow HTTP Response from "http://video.example.com/Sessions/Playing" to "172.17.0.1" in 0:00:01.4800168 with Status Code 204
[2020-08-31 19:41:39.025 +00:00] [INF] [22] Jellyfin.Api.Helpers.TranscodingJobHelper: Stopping ffmpeg process with q command for "/config/transcoding-temp/transcodes/0fea17ab23184294e79194c96147614c.m3u8"
[2020-08-31 19:41:39.052 +00:00] [INF] [22] Jellyfin.Api.Helpers.TranscodingJobHelper: FFMpeg exited with code 0
[2020-08-31 19:41:39.055 +00:00] [INF] [22] Jellyfin.Api.Helpers.TranscodingJobHelper: Deleting partial stream file(s) "/config/transcoding-temp/transcodes/0fea17ab23184294e79194c96147614c.m3u8"
[2020-08-31 19:41:39.164 +00:00] [WRN] [22] Emby.Server.Implementations.Networking.NetworkManager: "xx.xx.xx.xx/32": External request received, however, no WAN interface found.
[2020-08-31 19:41:40.637 +00:00] [INF] [18] Emby.Server.Implementations.Session.SessionManager: Playback stopped reported by app "Jellyfin Web" "10.7.0" playing "Genre". Stopped at "13802" ms`

xx.xx.xx.xx is my public ip
172.17.0.1 is the docker interface
/system/info/public gaves: 
`{"LocalAddress":"http://172.17.0.2:8096","ServerName":"jellyfin","Version":"10.7.0","ProductName":"Jellyfin Server","OperatingSystem":"Linux","Id":"c424b4cf1cc545a9bce6c3df21e49bfc"}`

**Comment by BaronGreenback** [comment]
Thanks - just looking to see how the chromecast detects devices.

**Comment by BaronGreenback** [comment]
I've think i have an idea of what's happening.

Where is your chromecast device located in your network? Is it in the xx.xx.xx.xx address range?

**Comment by BaronGreenback** [comment]
For those interested

https://www.cisco.com/c/en/us/td/docs/wireless/controller/technotes/7-6/chromecastDG76/ChromecastDG76.pdf
http://www.dial-multiscreen.org/dial-protocol-specification

Looks like one of the issue is that TTL is set to 1 for chromecast.

Will have a deeper read.

**Comment by idittansikte** [comment]
Why complicate stuff, just add two options to fill in public and local address and use them everywhere and disable the automatic stuff unless it's very local (like process to process on localhost). The automatic hostname/IP could be default if they are left out? You will always have ppl with "wierd" setups which will be hard to automate.

**Comment by sybnex** [comment]
strange thing, is still works with 10.6.1 ... 

i have to networks behind the router ... 192.168.x.x for LAN and another 192.168.y.x for WLAN. 
Jellyfin is located in the LAN (in docker) and the chromecast in the WLAN behind the LAN. 
The xx.xx.xx.xx is my real public IP address.

**Comment by BaronGreenback** [comment]
@sybnex How is the docker interface set up? Could you also post the startup settings which stipulate the LAN address range and interfaces?

**Comment by BaronGreenback** [comment]
I've testing a mod to the image. Will upload it tomorrow. If the problem is what is think it is - it should sort it.

**Comment by sybnex** [comment]
@BaronGreenback docker config is standard and running on a synology NAS (x86_64)

ip ad
`6: docker0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default 
    link/ether 02:42:77:53:3d:b0 brd ff:ff:ff:ff:ff:ff
    inet 172.17.0.1/16 brd 172.17.255.255 scope global docker0
       valid_lft forever preferred_lft forever`

docker info
`Server Version: 18.09.8
Storage Driver: btrfs
Logging Driver: db
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs db fluentd gcplogs gelf journald json-file local logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 0cf16177dbb234350dc27dd2bbd1d7cebd098108
runc version: 6cc9d3f2cd512eeb3d548e2f6b75bcdebc779d4d
init version: e01de58 (expected: fec3683)
Security Options:
 apparmor
Kernel Version: 4.4.59+
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 7.62GiB`

docker inspect jellyfin
`        "NetworkSettings": {
            "Bridge": "",
            "SandboxID": "69f9a059f80510657b35d988afc116ec6c42eb306d697a0d318f5a6850117857",
            "HairpinMode": false,
            "LinkLocalIPv6Address": "",
            "LinkLocalIPv6PrefixLen": 0,
            "Ports": {
                "8096/tcp": [
                    {
                        "HostIp": "0.0.0.0",
                        "HostPort": "8096"
                    }
                ]
            },
            "Gateway": "172.17.0.1",
            "GlobalIPv6Address": "",
            "GlobalIPv6PrefixLen": 0,
            "IPAddress": "172.17.0.2",
            "IPPrefixLen": 16,
            "IPv6Gateway": "",
            "MacAddress": "02:42:ac:11:00:02",
            "Networks": {
                "bridge": {
                    "Gateway": "172.17.0.1",
                    "IPAddress": "172.17.0.2",
                    "IPPrefixLen": 16,
                    "IPv6Gateway": "",
                    "GlobalIPv6Address": "",
                    "GlobalIPv6PrefixLen": 0,
                    "MacAddress": "02:42:ac:11:00:02",
                    "DriverOpts": null
                }
            }`

**Comment by BaronGreenback** [comment]
@sybnex I've updated the image - hopefully you should have more luck with chromecast now.

**Comment by roguedarkjedi** [comment]
@BaronGreenback where can I find this image that has the update? I don't see any difference in 10.6.4 uploaded like 6 days ago on GH.

**Comment by sybnex** [comment]
sorry for the late answer, but it still does not play on chromecast

`[18:53:21] [INF] [10] Jellyfin.Api.Helpers.TranscodingJobHelper: /usr/lib/jellyfin-ffmpeg/ffmpeg -fflags +genpts -i file:"/media/series/xxx.s01e01.mkv" -map_metadata -1 -map_chapters -1 -threads 0 -map 0:0 -map 0:1 -map -0:s -codec:v:0 copy -bsf:v h264_mp4toannexb -vsync -1 -codec:a:0 libmp3lame -ac 2 -ab 384000 -ar 48000 -af "volume=3" -copyts -avoid_negative_ts disabled -max_muxing_queue_size 2048 -f hls -max_delay 5000000 -hls_time 6 -individual_header_trailer 0 -hls_segment_type mpegts -start_number 0 -hls_segment_filename "/config/transcoding-temp/transcodes/637b650e453ba7271d1b877b67cb2aa4%d.ts" -hls_playlist_type vod -hls_list_size 0 -y "/config/transcoding-temp/transcodes/637b650e453ba7271d1b877b67cb2aa4.m3u8"
[18:53:22] [WRN] [22] Jellyfin.Server.Middleware.ResponseTimeMiddleware: Slow HTTP Response from http://video.example.com/videos/02c50b32-e3cb-aceb-9359-901f835b2287/hls1/main/0.ts?DeviceId=TW96aWxsYS81LjAgxMTsgTGludXggeDg2XzY0KSBBcsZVdlYktpdC81MzcuMgKEtIVE1MLCBsaWtlIY2tvKSBDaHJvbWUvODAuMCOTg3LjEwBTYWZhcmkvNTM3LjM2fDE1NDAyNjA0NjI1&MediaSourceId=02c50b32e3cbaceb9359901f835b2287&VideoCodec=h264&AudioCodec=mp3,aac&AudioStreamIndex=1&VideoBitrate=139552000&AudioBitrate=448000&PlaySessionId=ee38c59101d843d9bf5108e8bf7354c6&api_key=bb8e234dc9f74809bed6cb2091a6e9bf&TranscodingMaxAudioChannels=2&RequireAvc=false&Tag=b7b40349aaba999b059616f1dbfbcc6e&SegmentContainer=ts&MinSegments=1&BreakOnNonKeyFrames=True&h264-profile=high,main,baseline,constrainedbaseline,high10&h264-level=51&h264-deinterlace=true&TranscodeReasons=VideoCodecNotSupported to 172.17.0.1 in 0:00:01.137794 with Status Code 200
[18:53:24] [WRN] [22] Jellyfin.Server.Middleware.ResponseTimeMiddleware: Slow HTTP Response from http://video.example.com/Sessions/Playing/Progress to 172.17.0.1 in 0:00:00.6474862 with Status Code 204
[18:53:24] [WRN] [23] Jellyfin.Server.Middleware.ResponseTimeMiddleware: Slow HTTP Response from http://video.example.com/Sessions/Playing to 172.17.0.1 in 0:00:01.0278016 with Status Code 204
[18:53:27] [WRN] [18] Jellyfin.Server.Middleware.ResponseTimeMiddleware: Slow HTTP Response from http://video.example.com/Sessions/Playing/Progress to 172.17.0.1 in 0:00:00.6770934 with Status Code 204
[18:53:41] [INF] [20] Jellyfin.Api.Helpers.TranscodingJobHelper: Stopping ffmpeg process with q command for /config/transcoding-temp/transcodes/637b650e453ba7271d1b877b67cb2aa4.m3u8
[18:53:41] [WRN] [4] Emby.Server.Implementations.Networking.NetworkManager: xxx.xxx.xxx.xxx/32: External request received, however, no WAN interface found.
[18:53:41] [INF] [20] Jellyfin.Api.Helpers.TranscodingJobHelper: FFMpeg exited with code 0
[18:53:41] [INF] [20] Jellyfin.Api.Helpers.TranscodingJobHelper: Deleting partial stream file(s) /config/transcoding-temp/transcodes/637b650e453ba7271d1b877b67cb2aa4.m3u8
[18:53:43] [INF] [20] Emby.Server.Implementations.Session.SessionManager: Playback stopped reported by app Jellyfin Web 10.7.0 playing xxxxxxxxxxx Stopped at 17958 ms
[18:53:43] [WRN] [20] Jellyfin.Server.Middleware.ResponseTimeMiddleware: Slow HTTP Response from http://video.example.com/Sessions/Playing/Stopped to 172.17.0.1 in 0:00:01.8278611 with Status Code 204
`

btw: here my [network](https://mermaid.ink/img/eyJjb2RlIjoiZ3JhcGggVERcbiAgICBBW1JvdXRlcl0gLS0-fDE5Mi4xNjgueC4wfCBCXG4gICAgQihIQVByb3h5KSAtLT4gQ1xuICAgIEMoRG9ja2VyIG9uIE5BUykgLS0-fDE3Mi4xNy4wLjB8IEQoamVsbHlmaW4pXG4gICAgQSAtLT58MTkyLjE2OC55LjB8IEYoV0xBTilcbiAgICBGIC0tPiBHKENocm9tZUNhc3QpXG4gICAgICAgICAgICAiLCJtZXJtYWlkIjp7InRoZW1lIjoiZGVmYXVsdCIsInRoZW1lVmFyaWFibGVzIjp7ImJhY2tncm91bmQiOiJ3aGl0ZSIsInByaW1hcnlDb2xvciI6IiNFQ0VDRkYiLCJzZWNvbmRhcnlDb2xvciI6IiNmZmZmZGUiLCJ0ZXJ0aWFyeUNvbG9yIjoiaHNsKDgwLCAxMDAlLCA5Ni4yNzQ1MDk4MDM5JSkiLCJwcmltYXJ5Qm9yZGVyQ29sb3IiOiJoc2woMjQwLCA2MCUsIDg2LjI3NDUwOTgwMzklKSIsInNlY29uZGFyeUJvcmRlckNvbG9yIjoiaHNsKDYwLCA2MCUsIDgzLjUyOTQxMTc2NDclKSIsInRlcnRpYXJ5Qm9yZGVyQ29sb3IiOiJoc2woODAsIDYwJSwgODYuMjc0NTA5ODAzOSUpIiwicHJpbWFyeVRleHRDb2xvciI6IiMxMzEzMDAiLCJzZWNvbmRhcnlUZXh0Q29sb3IiOiIjMDAwMDIxIiwidGVydGlhcnlUZXh0Q29sb3IiOiJyZ2IoOS41MDAwMDAwMDAxLCA5LjUwMDAwMDAwMDEsIDkuNTAwMDAwMDAwMSkiLCJsaW5lQ29sb3IiOiIjMzMzMzMzIiwidGV4dENvbG9yIjoiIzMzMyIsIm1haW5Ca2ciOiIjRUNFQ0ZGIiwic2Vjb25kQmtnIjoiI2ZmZmZkZSIsImJvcmRlcjEiOiIjOTM3MERCIiwiYm9yZGVyMiI6IiNhYWFhMzMiLCJhcnJvd2hlYWRDb2xvciI6IiMzMzMzMzMiLCJmb250RmFtaWx5IjoiXCJ0cmVidWNoZXQgbXNcIiwgdmVyZGFuYSwgYXJpYWwiLCJmb250U2l6ZSI6IjE2cHgiLCJsYWJlbEJhY2tncm91bmQiOiIjZThlOGU4Iiwibm9kZUJrZyI6IiNFQ0VDRkYiLCJub2RlQm9yZGVyIjoiIzkzNzBEQiIsImNsdXN0ZXJCa2ciOiIjZmZmZmRlIiwiY2x1c3RlckJvcmRlciI6IiNhYWFhMzMiLCJkZWZhdWx0TGlua0NvbG9yIjoiIzMzMzMzMyIsInRpdGxlQ29sb3IiOiIjMzMzIiwiZWRnZUxhYmVsQmFja2dyb3VuZCI6IiNlOGU4ZTgiLCJhY3RvckJvcmRlciI6ImhzbCgyNTkuNjI2MTY4MjI0MywgNTkuNzc2NTM2MzEyOCUsIDg3LjkwMTk2MDc4NDMlKSIsImFjdG9yQmtnIjoiI0VDRUNGRiIsImFjdG9yVGV4dENvbG9yIjoiYmxhY2siLCJhY3RvckxpbmVDb2xvciI6ImdyZXkiLCJzaWduYWxDb2xvciI6IiMzMzMiLCJzaWduYWxUZXh0Q29sb3IiOiIjMzMzIiwibGFiZWxCb3hCa2dDb2xvciI6IiNFQ0VDRkYiLCJsYWJlbEJveEJvcmRlckNvbG9yIjoiaHNsKDI1OS42MjYxNjgyMjQzLCA1OS43NzY1MzYzMTI4JSwgODcuOTAxOTYwNzg0MyUpIiwibGFiZWxUZXh0Q29sb3IiOiJibGFjayIsImxvb3BUZXh0Q29sb3IiOiJibGFjayIsIm5vdGVCb3JkZXJDb2xvciI6IiNhYWFhMzMiLCJub3RlQmtnQ29sb3IiOiIjZmZmNWFkIiwibm90ZVRleHRDb2xvciI6ImJsYWNrIiwiYWN0aXZhdGlvbkJvcmRlckNvbG9yIjoiIzY2NiIsImFjdGl2YXRpb25Ca2dDb2xvciI6IiNmNGY0ZjQiLCJzZXF1ZW5jZU51bWJlckNvbG9yIjoid2hpdGUiLCJzZWN0aW9uQmtnQ29sb3IiOiJyZ2JhKDEwMiwgMTAyLCAyNTUsIDAuNDkpIiwiYWx0U2VjdGlvbkJrZ0NvbG9yIjoid2hpdGUiLCJzZWN0aW9uQmtnQ29sb3IyIjoiI2ZmZjQwMCIsInRhc2tCb3JkZXJDb2xvciI6IiM1MzRmYmMiLCJ0YXNrQmtnQ29sb3IiOiIjOGE5MGRkIiwidGFza1RleHRMaWdodENvbG9yIjoid2hpdGUiLCJ0YXNrVGV4dENvbG9yIjoid2hpdGUiLCJ0YXNrVGV4dERhcmtDb2xvciI6ImJsYWNrIiwidGFza1RleHRPdXRzaWRlQ29sb3IiOiJibGFjayIsInRhc2tUZXh0Q2xpY2thYmxlQ29sb3IiOiIjMDAzMTYzIiwiYWN0aXZlVGFza0JvcmRlckNvbG9yIjoiIzUzNGZiYyIsImFjdGl2ZVRhc2tCa2dDb2xvciI6IiNiZmM3ZmYiLCJncmlkQ29sb3IiOiJsaWdodGdyZXkiLCJkb25lVGFza0JrZ0NvbG9yIjoibGlnaHRncmV5IiwiZG9uZVRhc2tCb3JkZXJDb2xvciI6ImdyZXkiLCJjcml0Qm9yZGVyQ29sb3IiOiIjZmY4ODg4IiwiY3JpdEJrZ0NvbG9yIjoicmVkIiwidG9kYXlMaW5lQ29sb3IiOiJyZWQiLCJsYWJlbENvbG9yIjoiYmxhY2siLCJlcnJvckJrZ0NvbG9yIjoiIzU1MjIyMiIsImVycm9yVGV4dENvbG9yIjoiIzU1MjIyMiIsImNsYXNzVGV4dCI6IiMxMzEzMDAiLCJmaWxsVHlwZTAiOiIjRUNFQ0ZGIiwiZmlsbFR5cGUxIjoiI2ZmZmZkZSIsImZpbGxUeXBlMiI6ImhzbCgzMDQsIDEwMCUsIDk2LjI3NDUwOTgwMzklKSIsImZpbGxUeXBlMyI6ImhzbCgxMjQsIDEwMCUsIDkzLjUyOTQxMTc2NDclKSIsImZpbGxUeXBlNCI6ImhzbCgxNzYsIDEwMCUsIDk2LjI3NDUwOTgwMzklKSIsImZpbGxUeXBlNSI6ImhzbCgtNCwgMTAwJSwgOTMuNTI5NDExNzY0NyUpIiwiZmlsbFR5cGU2IjoiaHNsKDgsIDEwMCUsIDk2LjI3NDUwOTgwMzklKSIsImZpbGxUeXBlNyI6ImhzbCgxODgsIDEwMCUsIDkzLjUyOTQxMTc2NDclKSJ9fSwidXBkYXRlRWRpdG9yIjpmYWxzZX0)

**Comment by mholt** [comment]
Okay, update: a week later I'm still having this problem even with the web server hacks above. (Using 10.6.4 now.)

**Comment by dkanada** [comment]
@mholt what have you tried so far? Most people seem to have resolved the issue with LocalAddress modifications, but no one is completely sure that's the issue.

**Comment by mholt** [comment]
Well, as I've said, it's pretty flakey and I haven't been able to consistently & reliably reproduce the bug: I can't consistently experience it, and I apparently can't consistently fix it. I'll post here as I try new variations. But so far my experiments have consisted of what I've described: fiddling with restarting the service, installing or uninstalling the Intros (prerolls) plugin, etc... but no config changes. One week it works, the next week it doesn't.

Which log file should I be looking in that would have relevant information? Just `jellyfin*.log`?

**Comment by dkanada** [comment]
We only have two logs, one for system and the other for transcode jobs, so that's the right one.

**Comment by mholt** [comment]
Gotcha. Unfortunately, nothing appears to be added to the logs when Play is tapped. I will update here if I ever observe differently.

**Comment by dkanada** [comment]
It might be a frontend issue, Chromecast doesn't really use the server for anything.

**Comment by BaronGreenback** [comment]
@mholt - are you able to compile a PR, or do you run docker? If so, could you try out this build? It's the latest unstable build with a new backend interface. It should solve, or give us the info required to solve, the chromecast issue.

https://github.com/jellyfin/jellyfin/pull/3994

It has more network logging that should point to this issue if it doesn't work.

**Comment by mholt** [comment]
@BaronGreenback I don't use Docker, and I haven't compiled Jellyfin before, but I'd like to try that PR. I'm mobile atm, but do you have a quick link to instructions for compiling a PR? I haven't had a chance to look if this repo has them somewhere.

Also I'd be compiling on Mac for Linux. My Raspi is not powerful enough to compile...

**Comment by mholt** [comment]
@BaronGreenback Seems the build instructions are outdated. `./build ubuntu.armhf all` is not accepted, printing out help text instead.

According to `./build --help` I adjusted the command to be `./build -t native -p ubuntu.armhf` but I get this error:

```
% ./build -t native -p ubuntu.armhf
Cross-building is not supported for native builds, use 'docker' builds on amd64 for cross-building.
```

How can I cross-compile without Docker?

**Comment by tyhawk** [comment]
I'm also havinf issues with playback on the Chromecast. Whenever I try to cast, nothing happens. When I log in to Jellyfin on a browser (even the one on my LG tv) I can cast to that without any issue.

My setup is:
* Jellyfin 10.6.4 (jellyfin/jellyfin image)
* Docker Swarm on Ubuntu 18.04 (up-to-date).
* Traefik 2
* Traefik configured using the labels field in docker-compose.

I'm also not sure how to implement/test the Traefik config suggested by @Pegasystem within my setup (which doesn't use config files):
 
> For anyone Googling this and using Traefik 2, here's a .yml file to fix the issue:
> 
> ```
> http:
>   routers:
>     chromecast-rtr:
>       rule: "HostHeader(`your.jellyfin.url`) && Path(`/System/Info/Public`)"  # Replace with the proper URL
>       entryPoints:
>         - https
>       service: jellyfin-chromecast-local
>       priority: 100  # Set this to anything that's higher than your normal Jellyfin rule - if it doesn't have a priority, add one
>   services:
>     jellyfin-chromecast-local:
>       loadBalancer:
>         servers:
>           - url: "http://webserver-hosting-fixed-json-file/"  # Replace with the proper URL
> ```
> 
> Unfortunately Traefik isn't able to serve files on it's own from what I could find, so you'll have to host the modified json file somewhere. Be sure to put the file at `/System/Info/Public` on the webserver.

Does anyone have any suggestions on how to pull it off in config like
```
deploy:
      labels:
        - "traefik.enable=true"
        - "traefik.port=8096"
        - "traefik.http.services.jellyfin.loadbalancer.server.port=8096"
        - "traefik.http.middlewares.jellyfin-redirect-web-secure.redirectscheme.scheme=https"
        - "traefik.http.middlewares.jellyfin-redirect-web-secure2.redirectregex.permanent=true"
        - "traefik.http.middlewares.cloudHeader.headers.stsSeconds=63072000"
        - "traefik.http.middlewares.cloudHeader.headers.stsIncludeSubdomains=true"
        - "traefik.http.middlewares.cloudHeader.headers.stsPreload=true"
        - "traefik.http.routers.jellyfin-web.middlewares=cloud-redirect-web-secure"
        - "traefik.http.routers.jellyfin-web.rule=Host(`my.jellyfin.url`)"
        - "traefik.http.routers.jellyfin-web.entrypoints=web"
        - "traefik.http.routers.jellyfin-websecure.rule=Host(`my.jellyfin.url`)"
        - "traefik.http.routers.jellyfin-websecure.tls.certresolver=letsencryptresolver"
        - "traefik.http.routers.jellyfin-websecure.tls=true"
        - "traefik.http.routers.jellyfin-websecure.entrypoints=websecure"
```

**Comment by zakame** [comment]
I think it is probably worth noting that the problem isn't so much just on the presence of `LocalAddress` in the API response, but also in the jellyfin-chromecast app loaded:

![image](https://user-images.githubusercontent.com/110625/92922873-15a37a00-f469-11ea-8dbe-72f3e51f8604.png)

I currently upgraded from 10.6.1 to 10.6.4 after trying 10.6.2 previously and seeing this breakage, and I note that from 10.6.2 and above we load jellyfin-chromecast-3.0.0 and nightlies only, no longer the old 2.1.0 which can still be loaded from 10.6.1; this is noticeable by way that the old chromecast app can identify my device's name (Qchan) as seen above.  Reverting back to 10.6.1 and chromecast 2.1.0 restores functionality for now.

I also run jellyfin behind a reverse proxy (through nginx-ingress under k8s,) and can confirm working Chromecast functionality from 10.5.5 to 10.6.1.

**Comment by YouKnowBlom** [comment]
> I currently upgraded from 10.6.1 to 10.6.4 after trying 10.6.2 previously and seeing this breakage, and I note that from 10.6.2 and above we load jellyfin-chromecast-3.0.0 and nightlies only, no longer the old 2.1.0 which can still be loaded from 10.6.1

Strange, I don't think the Chromecast version should default to unstable. Are you able to switch back to stable under playback settings? If so does the old receiver work with 10.6.4?

**Comment by tyhawk** [comment]
I can confirm that with my setup (as mentioned a few comments ago) reverting the Jellyfin server container to version 10.6.1 solved the Chromecast playback issue. So it seems one of the changes in version 10.6.2 broke this functionality.

**Comment by mholt** [comment]
@tyhawk That's good to know, but I also experienced the problem with 10.6.0.

Specifically, the web UI / client says "MessagePlaybackErrorUndefined". Whatever that means 🙃

**Comment by zakame** [comment]
@YouKnowBlom I didn't change the Chromecast version under "User Settings > Playback" at all; it has always been set as Stable for me, which is why I noticed the sudden change of jellyfin-chromecast version as reported by the dashboard from 10.6.2 until 10.6.4; certainly strange since checking jellyfin-web/src/plugins/chromecastPlayer/plugin.js, it has no changes between v10.6.1...v10.6.4.

**Comment by YouKnowBlom** [comment]
@zakame My bad, I've been informed that the unstable version was moved to stable recently. It should be reverted back now

**Comment by fernandossmm** [comment]
Had this problem for a while, tried today after reading that last update and for me it's working correctly. More of an anecdote than a test, but if people arrive here you might find it's fixed now.

**Comment by zakame** [comment]
Rather than a workaround, I had the wild idea of improving my UPnP/DLNA capability for my k8s Jellyfin, so I applied these changes:

```yaml
kind: Service
spec:
  ports:
  ...
  - port: 1900
    targetPort: 1900
    protocol: UDP
  type: NodePort
---
kind: Deployment
spec:
  template:
    spec:
      hostNetwork: true
      ...
      containers:
        ports:
        - containerPort: 1900
          protocol: UDP
        ...
```

Combining this with reverting back to 10.6.4 (and with support from my local network gateway's UPnP) seems to have the side effect of fixing `/System/Info/Public` without having to recourse with ingress fiddling, only ensuring my domain name resolves to my node's IP (rather than the pod's:)

```sh
% curl https://jellyfin.my.domain/System/Info/Public      
{"LocalAddress":"http://192.168.1.254:8096","ServerName":"tachikoma","Version":"10.6.4","ProductName":"Jellyfin Server","OperatingSystem":"Linux","Id":"d03793edfbc54a1798899ebede2ff51c"}
```

A happy side effect of this is as UPnP/DLNA is enabled, I can now also stream other devices besides Chromecast, such as my old PS3 :tada:

**Comment by YouKnowBlom** [comment]
I believe there are two different issues in here, I pushed a fix the other day that should help the people who are able to start casting but unable to play any media with an error message along the lines of chromecastCommunicationError (jellyfin/jellyfin-chromecast#60). Next step is getting #4173 merged as that should fix HLS on the chromecast that was broken recently.

**Comment by Drol** [comment]
This worked for me. Set Server Address Settings - Base URL to: https://jellyfin.yourdomain.top/. It is not logical because it is suposed to a base url without a domain...

This makes /System/Info/Public return no Local Address.

`{
"ServerName": "ServerName",
"Version": "10.6.4",
"ProductName": "Jellyfin Server",
"OperatingSystem": "Linux",
"Id": "bef44b25ee8d4b3b9819d93ceea56acd"
}`

I don't know why it works without it but it does.

**Comment by tyhawk** [comment]
@Drol **I just tested your suggestion and it works for me (Docker container, Traefik reverse proxy).** I can finally use my Chromecast again.
So it seems there is no need to add Traefik settings and such workarounds. Fixable in 10 seconds flat!
Thanks, Drol. I owe you a beverage of your choice!

**Comment by sybnex** [comment]
@Drol it seems to work until a restart of the jellyfin server ...

After the reboot it sets a wrong baseURL in the config: "/https://jellyfin.yourdomain.top/" which leads into another error: 
`Microsoft.AspNetCore.Routing.Patterns.RoutePatternException: The route template separator character '/' cannot appear consecutively. It must be separated by either a parameter or a literal value.`

Because the servers sets the "/" in the front of the config.

But the idea is good! ... Maybe the Developers can fix it and don't set the / in the front!

**Comment by PrplHaz4** [comment]
> 
> 
> This worked for me. Set Server Address Settings - Base URL to: https://jellyfin.yourdomain.top/. It is not logical because it is suposed to a base url without a domain...
> 
> This makes /System/Info/Public return no Local Address.
> 
> `{ "ServerName": "ServerName", "Version": "10.6.4", "ProductName": "Jellyfin Server", "OperatingSystem": "Linux", "Id": "bef44b25ee8d4b3b9819d93ceea56acd" }`
> 
> I don't know why it works without it but it does.

The fact that this works is likely a bug - so I wouldn't plan on it working in 10.7 with the revamped network backend (#3994)...

**Comment by BaronGreenback** [comment]
It'll be crashing the url parsing functions - and may cause some issues with the API's.

It should work in #3394, but if anyone wants to double check then we can make sure it works out of the box.

**Comment by alexdelorenzo** [comment]
> This worked for me. Set Server Address Settings - Base URL to: https://jellyfin.yourdomain.top/. It is not logical because it is suposed to a base url without a domain...
> 
> This makes /System/Info/Public return no Local Address.
> 
> `{ "ServerName": "ServerName", "Version": "10.6.4", "ProductName": "Jellyfin Server", "OperatingSystem": "Linux", "Id": "bef44b25ee8d4b3b9819d93ceea56acd" }`
> 
> I don't know why it works without it but it does.

This is what fixed casting to Chromecasts for me on 10.6.4, unfortunately.

**Comment by Ruthhl3ss** [comment]
> > This worked for me. Set Server Address Settings - Base URL to: https://jellyfin.yourdomain.top/. It is not logical because it is suposed to a base url without a domain...
> > This makes /System/Info/Public return no Local Address.
> > `{ "ServerName": "ServerName", "Version": "10.6.4", "ProductName": "Jellyfin Server", "OperatingSystem": "Linux", "Id": "bef44b25ee8d4b3b9819d93ceea56acd" }`
> > I don't know why it works without it but it does.
> 
> This is what fixed casting to Chromecasts for me on 10.6.4, unfortunately.

This works for me! Thanks for posting!

**Comment by ThHanke** [comment]
@Drol this fix just works...thx

**Comment by ABotelho23** [comment]
Is this planned to be fixed for the next Jellyfin version? The Base URL fix did not resolve my issue, and reading this thread seems to indicate that the fix will not work for 10.7.X.

My setup has my Jellyfin not directly accessible to clients, only through a Caddy reverse proxy. My domain points to the Caddy reverse proxy, and resolves to the internal IP address when inside, and the external IP address when outside.

**Comment by rlue** [comment]
I just tried @Drol's fix, and it worked at first, but then the server started crashing on all subsequent requests. `docker logs jellyfin` shows the following error/stack trace:

```
Microsoft.AspNetCore.Routing.Patterns.RoutePatternException: The route template separator character '/' cannot appear consecutively. It must be separated by either a parameter or a literal value.
   at Microsoft.AspNetCore.Routing.Patterns.RoutePatternParser.Parse(String pattern)
   at Microsoft.AspNetCore.Routing.Patterns.RoutePatternFactory.Parse(String pattern)
   at Microsoft.AspNetCore.Mvc.Routing.ActionEndpointFactory.AddEndpoints(List`1 endpoints, HashSet`1 routeNames, ActionDescriptor action, IReadOnlyList`1 routes, IReadOnlyList`1 conventions, Boolean createInertEndpoints)
   at Microsoft.AspNetCore.Mvc.Routing.ControllerActionEndpointDataSource.CreateEndpoints(IReadOnlyList`1 actions, IReadOnlyList`1 conventions)
   at Microsoft.AspNetCore.Mvc.Routing.ActionEndpointDataSourceBase.UpdateEndpoints()
   at Microsoft.AspNetCore.Mvc.Routing.ActionEndpointDataSourceBase.Initialize()
   at Microsoft.AspNetCore.Mvc.Routing.ActionEndpointDataSourceBase.get_Endpoints()
   at Microsoft.AspNetCore.Routing.CompositeEndpointDataSource.<>c.<Initialize>b__15_0(EndpointDataSource d)
   at System.Linq.Enumerable.SelectManySingleSelectorIterator`2.ToArray()
   at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
   at Microsoft.AspNetCore.Routing.CompositeEndpointDataSource.Initialize()
   at Microsoft.AspNetCore.Routing.CompositeEndpointDataSource.GetChangeToken()
   at Microsoft.AspNetCore.Routing.DataSourceDependentCache`1.Initialize()
   at System.Threading.LazyInitializer.EnsureInitializedCore[T](T& target, Boolean& initialized, Object& syncLock, Func`1 valueFactory)
   at System.Threading.LazyInitializer.EnsureInitialized[T](T& target, Boolean& initialized, Object& syncLock, Func`1 valueFactory)
   at Microsoft.AspNetCore.Routing.Matching.DataSourceDependentMatcher..ctor(EndpointDataSource dataSource, Lifetime lifetime, Func`1 matcherBuilderFactory)
   at Microsoft.AspNetCore.Routing.Matching.DfaMatcherFactory.CreateMatcher(EndpointDataSource dataSource)
   at Microsoft.AspNetCore.Routing.EndpointRoutingMiddleware.InitializeCoreAsync()
--- End of stack trace from previous location where exception was thrown ---
```

Any ideas?

**Comment by sybnex** [comment]
@rlue this is what I said all the time. the fix will work only after the startup. afterwards you have to remove this config by editing an conf file im jellyfin to get it up and running again. 

It's time that it have to be fixed!

**Comment by joshuaboniface** [comment]
Interesting fix, that definitely points to the problem being that something here isn't appending the `server name` bit before the baseURL. Should help narrow it down hopefully.

**Comment by rlue** [comment]
Thanks @joshuaboniface. Also just wanted to say I really appreciate all the hard work you guys put into this. I use it every day!

**Comment by kieeps** [comment]
I had the problem that i could cast to chromecasts on my local network but not on remote sites, i did try to have the full to main in baseurl but no luck, but i did specify local network with 192.168.0.0/16 and that somehow made it work again, maby jellyfin assume everything to be local unless specified?

**Comment by Lightning2X** [comment]
Can confirm that the issue was local address for me too. @Drol 's fix did not work for me since it would prepend the / before the baseurl as another user said.

@mholt 's fix worked for me (thanks!) since I also had caddy. However I run a nested config so I had to adjust some stuff but the idea is the same.

**Comment by ne0phyte** [comment]
> Does anyone have any suggestions on how to pull it off in config like
> 
> ```
> deploy:
>       labels:
>         - "traefik.enable=true"
>         - "traefik.port=8096"
>         - "traefik.http.services.jellyfin.loadbalancer.server.port=8096"
>         - "traefik.http.middlewares.jellyfin-redirect-web-secure.redirectscheme.scheme=https"
>         - "traefik.http.middlewares.jellyfin-redirect-web-secure2.redirectregex.permanent=true"
>         - "traefik.http.middlewares.cloudHeader.headers.stsSeconds=63072000"
>         - "traefik.http.middlewares.cloudHeader.headers.stsIncludeSubdomains=true"
>         - "traefik.http.middlewares.cloudHeader.headers.stsPreload=true"
>         - "traefik.http.routers.jellyfin-web.middlewares=cloud-redirect-web-secure"
>         - "traefik.http.routers.jellyfin-web.rule=Host(`my.jellyfin.url`)"
>         - "traefik.http.routers.jellyfin-web.entrypoints=web"
>         - "traefik.http.routers.jellyfin-websecure.rule=Host(`my.jellyfin.url`)"
>         - "traefik.http.routers.jellyfin-websecure.tls.certresolver=letsencryptresolver"
>         - "traefik.http.routers.jellyfin-websecure.tls=true"
>         - "traefik.http.routers.jellyfin-websecure.entrypoints=websecure"
> ```

Apparently you can't route to external services via docker provider in traefik, but you can route to services in other containers.
My quick n dirty solution was adding an nginx container to the jellyfin docker-compose that only serves that file.
Create the fixed Public.json in the same directory as your jellyfin docker-compose.yml and add this service:
```
 nginx-chromecast-fix:
    image: nginx:alpine
    volumes:
      - ./Public.json:/usr/share/nginx/html/System/Info/Public:ro
    environment:
      - NGINX_PORT=80
    networks:
      - traefik
    labels:
      - "traefik.http.routers.nginx-chromecast-fix.service=nginx-chromecast-fix-svc"
      - "traefik.http.services.nginx-chromecast-fix-svc.loadBalancer.server.port=80"
```
and add these labels to the jellyfin container:
```
## chromecast fix
- "traefik.http.routers.jellyfin-chromecast-fix.entryPoints=https"
- "traefik.http.routers.jellyfin-chromecast-fix.rule=Host(`your.jellyfin.domain`) && Path(`/System/Info/Public`)"
- "traefik.http.routers.jellyfin-chromecast-fix.service=nginx-chromecast-fix-svc"
```

A bit wasteful for a single file but avoids creating an extra static config file for traefik.

**Comment by Dauliac** [comment]
> This worked for me. Set Server Address Settings - Base URL to: https://jellyfin.yourdomain.top/. It is not logical because it is suposed to a base url without a domain...
> 
> This makes /System/Info/Public return no Local Address.
> 
> `{ "ServerName": "ServerName", "Version": "10.6.4", "ProductName": "Jellyfin Server", "OperatingSystem": "Linux", "Id": "bef44b25ee8d4b3b9819d93ceea56acd" }`
> 
> I don't know why it works without it but it does.

I exactly have the same strange behavior. But that's broke  the dashboard :cry:

**Comment by YouKnowBlom** [comment]
This is a call for testing for anyone that is available: jellyfin/jellyfin-web#2085

**Comment by xsteadfastx** [comment]
i stepped into that problem and took a own hacky approach... container that grabs the origin `/System/Info/Public`, removes the `LocalAddress` and serves it back.

https://git.xsfx.dev/xsteadfastx/jellyfixer

**Comment by Koma-Andrea** [comment]
Looks like the rc is still affected

**Comment by idittansikte** [comment]
> Looks like the rc is still affected

Yes, I can't get it working either. Feels like it's never ending, been waiting months now for a proper fix. I assume chromecast is hard.

**Comment by ABotelho23** [comment]
I'm seeing the same thing here, **BUT** I actually get a message on the clients saying "MessageChromecastConnectionError", which I was not getting on 10.6.4 and lower. On 10.6.4 and lower it would just spin forever on the clients.

I tried `tail -f` on the server log while trying to cast, and it doesn't seem to spit out an error. Definitely a client-side error as far as I can tell from that.

**Comment by idittansikte** [comment]
You are right, I also got "MessageChromecastConnectionError" which seem to come from the chromecast (or libs interfacing with it on client side). The error does not come with any details in the javascript. I also get a few packets sent from chromecast to server when trying to play a movie, so it seem the chromecast is reaching the server at least. I also does not get any errors on server side.

**Comment by Koma-Andrea** [comment]
> i stepped into that problem and took a own hacky approach... container that grabs the origin `/System/Info/Public`, removes the `LocalAddress` and serves it back.
> 
> https://git.xsfx.dev/xsteadfastx/jellyfixer

You have a PR

**Comment by xsteadfastx** [comment]
> > i stepped into that problem and took a own hacky approach... container that grabs the origin `/System/Info/Public`, removes the `LocalAddress` and serves it back.
> > https://git.xsfx.dev/xsteadfastx/jellyfixer
> 
> You have a PR

reviewed it... one last question and i will merge and release :)

**Comment by Erriez** [comment]
I've `MessageChromecastConnectionError` when connecting to Chrome Cast as well and I'm using the following configuration on a Raspberry Pi 4:

**docker-compose.yml**
```yaml
version: '3'

services:
  jellyfin:
    image: ghcr.io/linuxserver/jellyfin
    container_name: jellyfin
    restart: always
    environment:
      - PUID=$DOCKER_UID
      - PGID=$DOCKER_GID
      - TZ=${TZ}
    volumes:
      - ${VOLUME_DIR}/jellyfin/Public.json:/usr/share/nginx/html/System/Info/Public:ro
      - ${VOLUME_DIR}/jellyfin/library:/config
      - ${VOLUME_DIR}/share/photos:/data/photos
      - ${VOLUME_DIR}/share/movies:/data/movies
      - ${VOLUME_DIR}/share/music:/data/music
      - /opt/vc/lib:/opt/vc/lib # Path for Raspberry Pi OpenMAX libs optional.
    ports:
      - 8096:8096 # Http webUI.
      # - 8920:8920 #optional
      - 7359:7359/udp # Optional - Allows clients to discover Jellyfin on the local network.
      - 1900:1900/udp # Optional - Service discovery used by DNLA and clients.

  swag:
    image: ghcr.io/linuxserver/swag
    container_name: swag
    restart: always
    ports:
      - 80:80
      - 443:443
    cap_add:
      - NET_ADMIN
    environment:
      - PUID=$DOCKER_UID
      - PGID=$DOCKER_GID
      - EMAIL=${SWAG_EMAIL}
      - URL=${ROOT_DOMAIN}
      - SUBDOMAINS=${SWAG_SUBDOMAINS}
      - VALIDATION=${SWAG_VALIDATION}
      - DNSPLUGIN=${SWAG_DNSPLUGIN}
      - PROPAGATION=${SWAG_PROPAGATION}
      - ONLY_SUBDOMAINS=${SWAG_ONLY_SUBDOMAINS}
      - TZ=${TZ}
    volumes:
      - ${VOLUME_DIR}/swag/config:/config
```

Does anyone have a suggestion to get Chrome Cast up and running?
